biothings/utils/configuration.py

```diff
@@ -245,9 +246,21 @@ def get_value_from_db(self, name):
         if not self._db:  # without db, only support module params.
             raise AttributeError("Transient parameter requires DB setup.")
 
-        doc = self._db.find_one({"_id": name})
+        doc = None
+        retry = 0
+        while True and retry < 30:
+            try:
+                doc = self._db.find_one({"_id": name})
+                if not doc:
+                    raise AttributeError(name)
+                retry += 1
+                break
+            except AutoReconnect:
+                self._db.__database.close()
+                self._db = self._get_db_function()
+
         if not doc:
-            raise AttributeError(name)
+            raise Exception("Cannot find document due to connection problem")
```

Comment:
@sengineer0 any more specific exception can be raised here?

Reply:
you meant we should define a custom Exception for this exception, like MaxRetryAutoReconnectException, do you?

Reply:
@sengineer0 Maybe first to check if there is a suitable exception from pymongo or Python's built-in exceptions. If not, yes, create a new exception is better than just use the generic Exception class.

Reply:
There is no suitable exception. I think it is better to create a custom one.